qutebrowser/components/braveadblock.py

```diff
@@ -330,6 +429,142 @@ def on_method_changed() -> None:
+            f"num_exceptions={len(cosmetic_resources.exceptions)}, "
+            f"len_injected_script={len(cosmetic_resources.injected_script)}, "
+            "generichide="
+            f"{cosmetic_resources.generichide})"  # type: ignore[attr-defined]
+        )
+
+        css = _make_css(cosmetic_resources)
+        num_css_lines = css.count("\n")
+        logger.debug(
+            f"Site-specific css on tab {tab.tab_id} for {url}: "
+            f"num_lines={num_css_lines}"
+        )
+
+        ad_blocker.cosmetic_map[tab.tab_id] = _CosmeticInfo(
+            cosmetic_resources.exceptions,
+            cosmetic_resources.generichide,  # type: ignore[attr-defined]
+        )
+
+        tab.remove_dynamic_css("adblock_generic")
+        tab.add_dynamic_css("adblock", css)
+
+        js_code = cosmetic_resources.injected_script
+        tab.remove_web_script("adblock")
+        num_js_lines = js_code.count("\n")
+        logger.debug(
+            f"Inject script on tab {tab.tab_id} for {url}: num_lines={num_js_lines}"
+        )
+
+        # We use the main world so the adblock scripts can interact with the page.
+        tab.add_web_script(
+            "adblock",
+            js_code,
+            world=usertypes.JsWorld.main,
+            injection_point=usertypes.InjectionPoint.creation,
+            subframes=True,
+        )
+
+
+@hook.load_finished()
+def update_cosmetic_filters(tab: apitypes.Tab, ok: bool) -> None:
+    """After loading a page, search for all classes and ids for generic selectors."""
+    if ad_blocker is None or not ok:
+        return
+
+    if tab.tab_id not in ad_blocker.cosmetic_map:
+        logger.info(
+            f"Tab {tab.tab_id} does not have an entry in the cosmetic map. "
+            "You probably need to run :adblock-update-resources"
+        )
+        return
+
+    assert tab.tab_id in ad_blocker.cosmetic_map
+    cosmetic_info = ad_blocker.cosmetic_map.pop(tab.tab_id)
+
+    # If generichide is True, that means we do not query for any additional generic
+    # rules
+    if cosmetic_info.generichide:
+        return
+
+    js_code = """{
+let classes = new Set();
+let ids = new Set();
+const walker = document.createTreeWalker(document.body);
+let node;
+while (node = walker.nextNode()) {
+    if (node.className) {
+        classes.add(node.className.toString());
+    }
+    if (node.id) {
+        ids.add(node.id.toString());
+    }
+}
+({"classes": Array.from(classes), "ids": Array.from(ids)})
+}
+"""
+
+    url = tab.url()
+
+    def _hidden_class_id_selectors_cb(data: Any) -> None:
+        logger.debug(
+            f"hidden_class_id_selectors_cb called on tab {tab.tab_id} for {url} "
+            f"with data type: {type(data)}"
+        )
+        if data is None:
+            return
+
+        assert isinstance(data, dict)
+        assert "classes" in data
+        assert "ids" in data
+
+        selectors = ad_blocker.hidden_class_id_selectors(  # type: ignore[union-attr]
```

Comment:
If the mypy linters are to be trusted, this can do without the tpye ignore comment.

```suggestion
        selectors = ad_blocker.hidden_class_id_selectors(
```